### PR TITLE
better check datetime dynamicfields

### PR DESCRIPTION
**Pull request by mamico**
A different implementation for better guessing datetime dynamicfields.

With previous method (endswith) I had problem when a textfield name is something like "candidate" and a datetime field as name "date".

### CODE REVIEW

**Comment by delijati** [comment]
Is this backward compatible?

**Comment by mamico** [comment]
I don't know about backward compatibility. But in my opinion this is a bug fix, not a new feature nor an improvement: current implementation manages wrongly Solr's schema configuration. (see test included).

File: scorched/connection.py [hunk 1]
```diff
@@ -286,9 +282,7 @@ def _prepare_docs(self, docs):
                 # fields
                 if value is None:
                     continue
-                if name in self._datefields:
```
Comment (delijati) [inline scorched/connection.py]: This should be still in. If a `name` is in `self._datefields` we already know its type `date`

File: scorched/response.py [hunk 2]
```diff
@@ -108,9 +108,7 @@ def from_json(cls, node, datefields=()):
     def _prepare_docs(self, docs, datefields):
         for doc in docs:
             for name, value in list(doc.items()):
-                if name in datefields:
```
Comment (delijati) [inline scorched/response.py]: Same as above, we already know the type.

File: scorched/connection.py (hunk 1 above)
Comment (mamico) [inline scorched/connection.py]: True. But this check was moved, not removed. See below first lines of is_datetime_field (lines 136, 137)

File: scorched/response.py (hunk 2 above)
Comment (mamico) [inline scorched/response.py]: True. But this check was moved, not removed. See above first lines of is_datetime_field (lines 136, 137)

Comment (delijati) [inline scorched/response.py]: Ah yeah right i oversaw this :)

**Pull request merged**